### ISSUE TITLE
Consider a flag on events to make them non-searchable

### ISSUE THREAD

**Issue by turt2live** [matrix-org/matrix-doc#2294]
A bot that posts a daily update sure does make it hard to search for things. */me glares at mscbot*

**Comment by Half-Shot** [comment]
Alternatively, we could just support richer filtering for search..

**Comment by turt2live** [comment]
That fixes one half of the problem. If I know that indexing something is a waste of time, I should be able to say "don't index this". Richer filtering just means the user spends more time filtering things out than they do finding things.

**Comment by Half-Shot** [comment]
Yep, I understand but I do question whether it should be up to the sender to decide if something should be indexed. I guess it could be a useful hint flag in the event's contents which some homeservers could ignore if they wanted. 

My reasoning for not wanting the flag is because I'm not sure I want others to dictate how my indexing works, but granted there is also a need to not index everything.

Regardless, richer filtering is certainly something that would be nice to have imo.

**Comment by turt2live** [comment]
I'd imagine it's more a "I'm a bot that would send a `m.notice` but my use case says that's not reasonable, so don't index this" thing (I'm somewhat confident that implementations don't currently index notices)
